src/build/genprod.scala

```diff
@@ -360,7 +360,7 @@ class Product(val i: Int) extends Group("Product") with Arity {
 
   def cases = {
     val xs = for ((x, i) <- mdefs.zipWithIndex) yield "case %d => %s".format(i, x)
-    val default = "case _ => throw new IndexOutOfBoundsException(n.toString())"
+    val default = "case _ => throw new IndexOutOfBoundsException(s\"$n is out of bounds (min 0, max " + (i-1) +")\")"
```

Comment:
I intentionally avoid string interpolation.

Because it causes sbt to raise `possible missing interpolator: detected an interpolated expression` error on startup (Similar to https://github.com/scala/bug/issues/9763)